RevenueCatUI/CustomerCenter/ViewModels/ManageSubscriptions/ManageSubscriptionsViewModel.swift

```diff
@@ -24,306 +24,33 @@ import SwiftUI
-            return path.type != .unknown
-        }
-    }
-
-}
-
-private extension Array<CustomerCenterConfigData.HelpPath> {
-    func relevantPathsForPurchase(
-        _ purchaseInformation: PurchaseInformation?
-    ) -> [CustomerCenterConfigData.HelpPath] {
-        guard let purchaseInformation else {
-            return self
-        }
-
-        return filter {
-            // if it's cancel, it cannot be a lifetime subscription
-            let isCancel = $0.type == .cancel
-            let isEligibleCancel = !purchaseInformation.isLifetime && !purchaseInformation.isCancelled
-
-            // if it's refundRequest, it cannot be free  nor within trial period
-            let isRefund = $0.type == .refundRequest
-            let isRefundEligible = purchaseInformation.price != .free
-                                    && !purchaseInformation.isTrial
-                                    && !purchaseInformation.isCancelled
-
-            // if it has a refundDuration, check it's still valid
-            let refundWindowIsValid = $0.refundWindowDuration?.isWithin(purchaseInformation) ?? true
-
-            return (!isCancel || isEligibleCancel) &&
-                    (!isRefund || isRefundEligible) &&
-                    refundWindowIsValid
-        }
-    }
-}
-
-private extension CustomerCenterConfigData.HelpPath.RefundWindowDuration {
-    func isWithin(_ purchaseInformation: PurchaseInformation) -> Bool {
-        switch self {
-        case .forever:
-            return true
-
-        case let .duration(duration):
-            return duration.isWithin(
-                from: purchaseInformation.latestPurchaseDate,
-                now: purchaseInformation.customerInfoRequestedDate
+            self.purchasesActive = purchasesActive
+            super.init(
+                screen: screen,
+                actionWrapper: actionWrapper,
+                purchaseInformation: nil,
+                refundRequestStatus: refundRequestStatus,
+                purchasesProvider: purchasesProvider,
+                loadPromotionalOfferUseCase: loadPromotionalOfferUseCase
             )
-
-        @unknown default:
-            return true
         }
-    }
-}
-
-private extension ISODuration {
-    func isWithin(from startDate: Date?, now: Date) -> Bool {
-        guard let startDate else {
-            return true
-        }
-
-        var dateComponents = DateComponents()
-        dateComponents.year = self.years
-        dateComponents.month = self.months
-        dateComponents.weekOfYear = self.weeks
-        dateComponents.day = self.days
-        dateComponents.hour = self.hours
-        dateComponents.minute = self.minutes
-        dateComponents.second = self.seconds
-
-        let calendar = Calendar.current
-        let endDate = calendar.date(byAdding: dateComponents, to: startDate) ?? startDate
 
-        return startDate < endDate && now <= endDate
+    func updatePurchases(_ purchasesActive: [PurchaseInformation]) {
```

Comment:
what's this

Reply:
This is something we introduced with @ajpallares 
In order to reload the list whenever `CustomerCenterViewModel` reloads `CustomerInfo`, we need to trigger an update of the `Published` property manually. That's triggered with an `onChange` in `ManageSubscriptionsView`